File: reservations/src/main/java/raf/rs/reservations/domain/Restaurant.java
package raf.rs.reservations.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;
import org.hibernate.annotations.ColumnDefault;

@Entity
@jakarta.persistence.Table(name = "restaurant")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Restaurant {

    @Id
    @GeneratedValue
    private Long id;

    private String name;

    private String address;

    private String description;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "restaurant", orphanRemoval = true)
    private List<Table> tables = new ArrayList<>();

    private String openTime;

    private String type;

    @ColumnDefault("-1")
    private int discountAfterXReservations;

    @ColumnDefault("-1")
    private int freeItemEachXReservations;

    private Long managerId;

}

File: frontend/src/types/restaurant.ts
import Table from "./table"

interface Restaurant {
    id: number,
    name: string,
    address: string,
    description: string,
    openTime: string,
    type: string,
    tables: Table[],
    discountAfterXReservations: number,
    freeItemEachXReservations: number,
    managerId: number
}

export default Restaurant